### DISCUSSION TITLE
sandpack-client Navigation

### DISCUSSION THREAD

**Discussion by babur001** [codesandbox/sandpack#368] [Q&A]
I am searching for a week now for an answer, the code does not work:

```js
client.dispatch({
   type: "urlchange",
   url: "/a.html",
});
```

Does not seem to change url, how is it done? Can you show in more detail or the source code where i can get an idea, thanks

**Comment by danilowoz** [comment]
Hey @SaburovBabur, thanks for raising this question.

The safest (and quickest) way to it is changing the iframe URL, for example we use the same approach on sandpack-react:
```js
const handleNewURL = (newUrl: string): void => {
  if (!iframeRef.current) {
    return;
  }

  iframeRef.current.src = newUrl;
};
```
https://github.com/codesandbox/sandpack/blob/main/sandpack-react/src/components/Preview/index.tsx#L93-L99

Probably, there are many different ways to solve it, but for while that's the way we recommend. Maybe in the future, we can approach this problem from a different perspective - feel free to open a feature request.

Hope it answers your question 😄
